engine/classes/ElggFile.php

```diff
@@ -80,6 +80,18 @@ public function getFilenameOnFilestore() {
 	}
 
 	/**
+	 * Returns path relative to the filestore (data directory) root
+	 *
```

Comment:
Custom filestores means this doesn't guarantee the path will be relative to `dataroot`. Maybe this API isn't such a good idea.